File: FinalProject/app/src/main/java/com/example/finalproject/items/Item.java
package com.example.finalproject.items;

import org.xmlpull.v1.XmlPullParser;

import java.io.Serializable;

public class Item implements Serializable {
    private int costSell, costBuy, rarity, category;
    private String name;

    public Item(){}

    public Item(String name, int costSell, int costBuy, int rarity, int category) {
        this.costSell = costSell;
        this.costBuy = costBuy;
        this.rarity = rarity;
        this.category = category;
        this.name = name;
    }

    public int getCostSell() {
        return costSell;
    }

    public void setCostSell(int costSell) {
        this.costSell = costSell;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getRarity() {
        return rarity;
    }

    public void setRarity(int rarity) {
        this.rarity = rarity;
    }

    public int getCategory() {
        return category;
    }

    public void setCategory(int category) {
        this.category = category;
    }

    public int getCostBuy() {
        return costBuy;
    }

    public void setCostBuy(int costBuy) {
        this.costBuy = costBuy;
    }
}

File: FinalProject/app/src/main/java/com/example/finalproject/items/Item.kt
package com.example.finalproject.items

open class Item(val costSell:Double, val costBuy:Double, val rarity:Int, val category:Int, val name:String) {

    constructor():this(0.0, 0.0, 0, 0, "")

    constructor(item: Item):this(item.costSell, item.costBuy, item.rarity, item.category, item.name)

    open fun clone():Item{
        return Item(costSell, costBuy, rarity, category, name)
    }
}